blockstore/api.py

```diff
@@ -0,0 +1,6 @@
+"""
+The Blockstore Python API.
+
+Imported here to allow consumers to import the public api from `blockstore.api`.
+"""
+from blockstore.apps.api import *  # pylint: disable=wildcard-import
```

Comment:
Is this necessary to do? These API calls return models that echo models that are in blockstore itself. Also, it's inconsistent with more recent practice in [OEP-49](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0049-django-app-patterns.html#data-py) to create a separate set of `attrs` data structures to pass back in these situations.

I realize this is an intermediate state, and I imagine a lot of the models brought over from edx-platform's client-side code. My main concern is that we'd be introducing an api module with some implied level of stability, and then possibly changing all of it.

Is the plan for what happens to the API redundancies written down somewhere? I'm fine with this as an intermediate state, but I'd like to understand where we're going with collapsing these down?

Reply:
@ormsbee 

I agree that eventually we should only have one set of `attrs`. However, the problem is that there are differences between the two sets of attrs. The easiest way to allow fast rollbacks and to keep the changes easy to understand was to keep them all in the `methods.api` file. If we try to collapse them now, it would require making changes to callers in various places in edx-platform and make rollbacks harder. This way ensures that from the point of view of callers, the API hasn't changed. And once blockstore is an in-process app, we can incrementally update the APIs and their callers.

> My main concern is that we'd be introducing an api module with some implied level of stability, and then possibly changing all of it.

Agree with this. @jvdm We should remove this file for now and only add methods later that we want to mark stable.

> Is the plan for what happens to the API redundancies written down somewhere? I'm fine with this as an intermediate state, but I'd like to understand where we're going with collapsing these down?

Nope, not yet. I was thinking we would look at the usages in edx-platform and then see how the APIs should be updated for better performance and ergonomics – the system boundary has changed and there may be opportunities for these?

Reply:
> @jvdm We should remove this file for now and only add methods later that we want to mark stable.

+1

Reply:
>  I was thinking we would look at the usages in edx-platform and then see how the APIs should be updated for better performance and ergonomics – the system boundary has changed and there may be opportunities for these?

Sounds good.